File: project/modules/tests/src/test/scala/rws/tests/utils/VoidEventsListener.scala
package rws.tests.utils;

import java.nio.ByteBuffer;

import rws.common.responsiveWebSocketConnection.api.{ResponsiveWsConnection => Rwsc};
import rws.common.responsiveWebSocketConnection.api.ResponsiveWsConnection.EventsListener;
import rws.common.responsiveWebSocketConnection.api.ResponseSender;

class VoidEventsListener extends EventsListener {
  override def onClose(c: Rwsc, code: Int, reason: String): Unit = {}
  override def onError(c: Rwsc, error: Throwable): Unit = {}

  override def onMalformedBinaryMessage(c: Rwsc, message: ByteBuffer): Unit = {}
  override def onTextMessage(c: Rwsc, message: String): Unit = {}

  override def onBinaryRequest(
    c: Rwsc,
    messageWithHeader: ByteBuffer,
    startIndex: Int,
    responseSender: ResponseSender
  ): Unit = {}

  override def onUnrequestingBinaryMessage(c: Rwsc, messageWithHeader: ByteBuffer, startIndex: Int): Unit = {}
}

File: project/modules/responsiveWebSocketConnection/src/main/java/rws/common/responsiveWebSocketConnection/impl/components/VoidEventsListener.java
package rws.common.responsiveWebSocketConnection.impl;

import java.nio.ByteBuffer;

import rws.common.responsiveWebSocketConnection.api.ResponsiveWsConnection;
import rws.common.responsiveWebSocketConnection.api.ResponseSender;

final class VoidEventsListener implements ResponsiveWsConnection.EventsListener {
  public static final VoidEventsListener instance = new VoidEventsListener();

  private VoidEventsListener() {}

  @Override
  public void onClose(ResponsiveWsConnection connection, int code, String reason, boolean isRemote) {}
  @Override
  public void onError(ResponsiveWsConnection connection, Throwable error) {}
  @Override
  public void onMalformedBinaryMessage(ResponsiveWsConnection connection, ByteBuffer message) {}
  @Override
  public void onTextMessage(ResponsiveWsConnection connection, String message) {}

  @Override
  public void onBinaryRequest(
    ResponsiveWsConnection connection,
    ByteBuffer messageWithHeader,
    int startIndex,
    ResponseSender responseSender
  ) {}

  @Override
  public void onUnrequestingBinaryMessage(
    ResponsiveWsConnection connection,
    ByteBuffer messageWithHeader,
    int startIndex
  ) {}
}
